src/main/java/org/mockito/internal/creation/bytebuddy/InlineDelegateByteBuddyMockMaker.java

```diff
@@ -108,103 +109,111 @@ class InlineDelegateByteBuddyMockMaker
+                                        + "which means this operation will fail to complete. Please first make sure you are not clearing this "
+                                        + "property anywhere, and failing that, raise a bug with your JVM vendor."));
+            }
 
-        try {
             try {
-                instrumentation = ByteBuddyAgent.install();
-                if (!instrumentation.isRetransformClassesSupported()) {
-                    throw new IllegalStateException(
-                            join(
-                                    "Byte Buddy requires retransformation for creating inline mocks. This feature is unavailable on the current VM.",
-                                    "",
-                                    "You cannot use this mock maker on this VM"));
-                }
-                File boot = File.createTempFile("mockitoboot", ".jar");
-                boot.deleteOnExit();
-                JarOutputStream outputStream = new JarOutputStream(new FileOutputStream(boot));
                 try {
-                    String source =
-                            "org/mockito/internal/creation/bytebuddy/inject/MockMethodDispatcher";
-                    InputStream inputStream =
-                            InlineDelegateByteBuddyMockMaker.class
-                                    .getClassLoader()
-                                    .getResourceAsStream(source + ".raw");
-                    if (inputStream == null) {
+                    instrumentation = ByteBuddyAgent.install();
+                    if (!instrumentation.isRetransformClassesSupported()) {
                         throw new IllegalStateException(
                                 join(
-                                        "The MockMethodDispatcher class file is not locatable: "
-                                                + source
-                                                + ".raw",
+                                        "Byte Buddy requires retransformation for creating inline mocks. This feature is unavailable on the current VM.",
                                         "",
-                                        "The class loader responsible for looking up the resource: "
-                                                + InlineDelegateByteBuddyMockMaker.class
-                                                        .getClassLoader()));
+                                        "You cannot use this mock maker on this VM"));
                     }
-                    outputStream.putNextEntry(new JarEntry(source + ".class"));
+                    File boot = File.createTempFile("mockitoboot", ".jar");
+                    boot.deleteOnExit();
+                    JarOutputStream outputStream = new JarOutputStream(new FileOutputStream(boot));
                     try {
```

Comment:
**suggestion:** I wonder if we can use the try with resources there, something like

```
try (JarOutputStream outputStream = new JarOutputStream(new FileOutputStream(boot))) {
```